### PR TITLE
BZ-1316966 - OSGi: drools-persistence-jpa throws NPE when starting a process via RuntimeManager

### PR DESCRIPTION
**Pull request by mswiderski**
@winklerm could you please take a look at this? Is this what you had in mind?

### CODE REVIEW

**Comment by psiroky** [comment]
I will take a look at his tomorrow (hopefully).

**Comment by winklerm** [comment]
Looks good to me.

I have extended the test case a bit and fixed issue with `kie-aries-blueprint/src/main/java/org/kie/aries/blueprint/namespace/KieEnvironmentElementParser`, please see https://github.com/mswiderski/droolsjbpm-integration/pull/5 and include it, if possible.

**Comment by mswiderski** [comment]
changes from @winklerm incorporated into this PR

**Comment by mswiderski** [comment]
@psiroky any comments on this?

File: drools-osgi/drools-karaf-itests/src/test/java/org/drools/karaf/itest/beans/AbstractProcessWithPersistenceBean.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
+package org.drools.karaf.itest.beans;
```
Comment (psiroky) [inline drools-osgi/drools-karaf-itests/src/test/java/org/drools/karaf/itest/beans/AbstractProcessWithPersistenceBean.java]: There is a missing license header here...

File: drools-osgi/drools-karaf-itests/src/test/java/org/drools/karaf/itest/beans/ProcessWithPersistenceDirectBean.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
+package org.drools.karaf.itest.beans;
```
Comment (psiroky) [inline drools-osgi/drools-karaf-itests/src/test/java/org/drools/karaf/itest/beans/ProcessWithPersistenceDirectBean.java]: License header as well.

File: drools-osgi/drools-karaf-itests/src/test/java/org/drools/karaf/itest/beans/ProcessWithPersistenceEnvBean.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
+package org.drools.karaf.itest.beans;
```
Comment (psiroky) [inline drools-osgi/drools-karaf-itests/src/test/java/org/drools/karaf/itest/beans/ProcessWithPersistenceEnvBean.java]: License header...

File: drools-osgi/drools-karaf-itests/src/test/java/org/drools/karaf/itest/blueprint/KieBlueprintjBPMPersistenceKarafIntegrationTest.java [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+                // Don't bother with local console output as it just ends up cluttering the logs
+                configureConsole().ignoreLocalConsole(),
+                // Force the log level to INFO so we have more details during the test.  It defaults to WARN.
+                logLevel(LogLevelOption.LogLevel.INFO),
+
+                // Option to be used to do remote debugging
+//                  debugConfiguration("5005", true),
+
+                // Load KIE features
+                loadKieFeatures("jndi", "transaction", "droolsjbpm-hibernate", "h2", "jbpm", "kie-aries-blueprint"),
```
Comment (psiroky) [inline drools-osgi/drools-karaf-itests/src/test/java/org/drools/karaf/itest/blueprint/KieBlueprintjBPMPersistenceKarafIntegrationTest.java]: Do we have these features: `jndi` and `transaction`? I could not find those. Am I overlooking something?

**Comment by psiroky** [comment]
Just a few minor comments, otherwise looks good.

**Comment by mswiderski** [comment]
fixed license headers and merged to master, thanks @winklerm @psiroky

**Pull request closed without merging**